flipperdevices/flipperzero-firmware
# [FL-1472, FL-1596, FL-1673] IRDA: stability improvements

# What's new

- Restrict with 31 bytes length for remote and signal name
- Don't stuck for 0 PWM cycle timings
- Support timings > 65535 PWM cycles
- Fix remote file open error
- Add IRDA TX debug redirect
- Add remote parse error print, improve parsing, support tabs
- Fix stucks with uncorrect RAW signal values, long strings in remote file, etc
- Fix HAL signals capturing (save previous read value)
- Fix leak in case of failed parsing

# Verification 

- try to rename button, remote with long name
- try to send long and short timings on any frequencies in RAW format
- try different signals saving to file and send them
- check leaks in all error cases

# Checklist (do not modify)
